kinoml/modeling/homology.py

```diff
@@ -66,178 +69,51 @@ def get_sequence(
         from io import StringIO
 
         if kinase:
-            if backend == "uniprot":
-                sequence = KinaseDomainAminoAcidSequence.from_uniprot(identifier)
-            elif backend == "ncbi":
-                sequence = KinaseDomainAminoAcidSequence.from_ncbi(identifier)
+            try:
+                from_method = getattr(KinaseDomainAminoAcidSequence, f"from_{backend}")
+            except AttributeError:
+                # TODO implement ncbi backend
+                if backend == "ncbi":
+                    raise NotImplementedError
+                else:
+                    raise ValueError(
+                        'Backend "%s" not supported. Please choose from ["uniprot", "ncbi"]'
+                        % (backend)
```

Comment:
`%`-style formatting is only used in the `logging` library now. Please use `f-strings` since we are only supporting Python 3.6+ anyway!